plugins/s3-viewer-backend/src/credentials-provider/RadosGwCredentialsProvider.ts

```diff
@@ -1,7 +1,7 @@
 import { Config } from '@backstage/config';
-import { SignatureV4 } from '@aws-sdk/signature-v4';
+import { SignatureV4 } from '@smithy/signature-v4';
```

Comment:
This change looked very odd to me, but as pointed out by the package itself (https://www.npmjs.com/package/@aws-sdk/signature-v4 & https://github.com/aws/aws-sdk-js-v3/blob/de8d4a6bcffdb94f0d3b23f65fdb203136498a31/deprecated/packages/signature-v4/README.md apparently this `@smithy/...` thing is the replacement.

I do find it a bit strange that something security-critical like these signatures get moved to a different package, but it looks legit to me.  It gets a bit odder even because https://www.npmjs.com/package/@smithy/signature-v4 states the following:

> It is internal to Smithy-TypeScript generated clients, and not generally intended for standalone usage outside this context.

Which is evidently not true anymore.

And finally "Smithy" itself seems to be an AWS project to write a new language for "defining services and SDKs": https://smithy.io/2.0/index.html

Reply:
Yes. I don't remember when exactly, but some time ago all those "smithy" dependencies were added in our yarn.lock. To me, it seems like they are migrating some internal things to be now part of "smithy", but as you discovered, the documentation might be still a bit out-of-date